pymc3/tests/test_distributions.py

```diff
@@ -1475,27 +1505,39 @@ def test_constantdist(self):
         self.check_logp(Constant, I, {"c": I}, lambda value, c: np.log(c == value))
 
     # Too lazy to propagate decimal parameter through the whole chain of deps
-    @pytest.mark.xfail(condition=(aesara.config.floatX == "float32"), reason="Fails on float32")
-    def test_zeroinflatedpoisson(self):
+    @pytest.mark.xfail(
+        condition=(aesara.config.floatX == "float32"),
+        reason="Fails on float32 due to inf issues",
+    )
+    # this test doesn't always result in the same outcome (pass/not pass)
```

Comment:
This one should always pass or fail, since it has no random subsampling... Are you sure?

Reply:
I haven't looked into the test logic but it seems to return both pass and fails on my machine. I will investigate

x-special/nautilus-clipboard
copy
file:///home/matteo/Pictures/Screenshot%20from%202021-03-03%2010-29-26.png

Reply:
You are right. I see now that `checkd` calls `check_int_to_1`, which has an `n_samples=10` in it. So the same source of randomness as the other tests. Thanks for checking

Reply:
We should also temporarily change this one inplace for n_samples=-1, in case we removed any xfail over a checkd distribution. Check it's fine and then put it back to 10. The relevant line is here: https://github.com/pymc-devs/pymc3/blob/ea1b03811f7d58a38c09a56a61144294f3732d71/pymc3/tests/test_distributions.py#L775